src/modules/authentication/interfaces/IClaimsChallenge.ts

```diff
@@ -0,0 +1,5 @@
+import { IQuery } from '../../../types/query-runner';
+
+export default interface IClaimsChallenge {
+  handleClaimsChallenge(responseHeaders: Headers, sampleQuery: IQuery): void;
```

Comment:
Nit: Since the class ClaimsChallenge is named without ambiguity, we may not need to have the exposed function called handleClaimsChallenge. It can still be clear if we say `handle` or `invoke`